### DISCUSSION TITLE
Reminders for events that are not "todos"

### DISCUSSION THREAD

**Discussion by Stuntflyer** [mikeberger/borg_calendar#173] [General]
When I enter a "todo" I can get a pop up reminder. When I don't set the event as a "todo", it seems like I don't get a reminder. Can someone please clarify this for me?

**Comment by mikeberger** [comment]
events with no time do not get reminders. events with specific times get popups as shown in the popup reminders section of the appointment editor.

**Reply by Stuntflyer** [reply]
I set a time and got one reminder. Based on the settings I thought I would get a reminder every 30 min. When the pop up occured I closed the window without checking anything off. What should I have done to get another reminder every 30 min?

![Capture](https://user-images.githubusercontent.com/46932988/194083060-f41dc45c-baed-475d-a324-c97f27169471.PNG)

**Reply by mikeberger** [reply]
I will assume that you are using the default option to have a list of reminders in one windows as opposed to the multi-window "post-it-note" style of reminders.

The 30 minute nagging reminders are for todos. Timed appointments have the user-defined popup schedule rather than the 30 minute one.

For timed appointments, the list pops up at the times you have set. If you have lots of items on the list, the list will pop up each time an appointment popup time requires it.

The reminder will go away when you are outside the range of times for the appointment. If you have only set 2880 minutes, you get your 2880 minute popup and then it will disappear from the list at some point when the list refreshes.

If you set times of 2880 minutes and also 5 minutes (for example), the appointment will remain on the list until from 2880 minutes, until after the 5 minute timer goes off. The appointment stays on the list over the range of times that you set. The times you set just trigger the popup window.

**Reply by Stuntflyer** [reply]
So, it seems that the best way to get 30 min reminders is to use a Todo. Is there a way to cancel the 30 min reminder while keeping the event on the calendar, without using a strike through?

**Reply by mikeberger** [reply]
Select the reminder in the list and press the Hide button to get a reminder off the list.

The latest committed code has a "quiet" button on the reminder list that stops all popping up, but keep the list. This is for when you want the list to stay on screen, but don't want the annoying popping up behavior while doing other stuff

**Comment by mikeberger** [comment]
As an FYI, since this is a discussion thread... - the reminder list functionality was probably coded based on requirements-by-committee  from lots of users on sourceforge. I don't remember exactly. It is now some of the most difficult code to understand in BORG and is overly complex. It was long ago, before other calendar programs existed to pattern it after. 

One very poor decision was to keep the previous post-it note reminder system instead of replacing it. the overly complex post-it note system was written in the early 90s on a SUN workstation before PCs were a thing. I was just making things up. Much of the weirdness of the reminder list is due to the old post-it note code being retained.

It all works, but I can no longer remember WHY it does what it does.

If it were coded from scratch today, it would just be a clone of google or outlook reminders.

**Reply by Stuntflyer** [reply]
Thank you, Mike
